### ISSUE TITLE
"Request timed out" messages in Chrome

### ISSUE THREAD

**Issue by baukezwaan** [silverstripe/silverstripe-mfa#414]
We have severe issues with users hitting concurring timeouts while validating their 6-digit code. It seems to be Chromium only. 

We checked the cache headers of the /Security/login/default/mfa page, but they seem to be correct (no-cache, no-store, must-revalidate, max-age=604800) and the Network Panel shows the page is not loaded from disk-cache. However, the page continues to be cached in Chrome. When opening Incognito Browser there is no problem logging in with MFA. Until you logout, and try to login again...

This is off course is very frustrating for some of our customers. Hopefully some one recognizes these symptoms and can offer some direction to solutions.

![image](https://user-images.githubusercontent.com/795238/102506698-acdd2680-4083-11eb-8f41-9ffcdebda091.png)

**Comment by Cheddam** [comment]
Possibly related to #393 - it looks like the CSRF token is invalid, likely due to a cached response from the schema API. Surprised to see this impact a modern browser, though - I couldn't reproduce the original issue anywhere outside of IE 11.

Can you review the headers in the `/Security/login/default/mfa/schema` request when the MFA UI is loaded, and see if you can identify any issues?

**Comment by baukezwaan** [comment]
We have inspected the network panel and, indeed; the /schema seems to be loaded from disk-cache! Although the cache-control headers are explicitly set to private, must-revalidate. 

```
Request URL: https://xxxxxxxxxxxxxxxxxxxxxx/Security/login/default/mfa/schema
Request Method: GET
Status Code: 200  (from disk cache)
Remote Address: 0.0.0.0.0:443
Referrer Policy: strict-origin-when-cross-origin
Response Headers
cache-control: private, must-revalidate, max-age=604800
content-length: 3389
content-type: application/json
date: Thu, 17 Dec 2020 15:04:34 GMT
etag: "f29c55a2ad648e77888fdff32aa9db63"
expires: Thu, 24 Dec 2020 15:04:34 GMT
server: Apache
vary: X-Forwarded-Protocol
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-ratelimit-limit: 10
x-ratelimit-remaining: 9
x-ratelimit-reset: 1608217475
x-xss-protection: 1; mode=block
Request Headers
Provisional headers are shown. Disable cache to see full headers.
Referer: https://xxxxxxxxxxxxxxxxxxxxxxxxx/Security/login/default/mfa
sec-ch-ua: "Google Chrome";v="87", " Not;A Brand";v="99", "Chromium";v="87"
sec-ch-ua-mobile: ?0
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
```

Probably the culprit is the endpoint `/Security/login/default/mfa/verify/totp` with the SecurityID in it. We now have temporary added anti-cache-headers in this response:

vendor/Silverstripe/mfa/src/RequestHandler/VerificationHandlerTrait.php:77
```
//...
$data[$token->getName()] = $token->getValue();

// -- start edit
// add extra headers
$response->addHeader('Pragma', 'no-cache');
$response->addHeader('Cache-Control', 'no-store, no-cache, must-revalidate, max-age=0');        
// -- end edit

// Respond with our method
return $response->setBody(json_encode($data));
```

**Comment by Cheddam** [comment]
Ah, excellent catch - would you mind raising that fix as a PR? 🙂

**Comment by emteknetnz** [comment]
Fixed in https://github.com/silverstripe/silverstripe-mfa/pull/417

**Comment by Cheddam** [comment]
@baukezwaan This should be resolved as of `silverstripe/mfa 4.2.2` - please let us know if you continue to experience issues after updating.
